piq/ssim.py

```diff
@@ -625,46 +546,61 @@ def _ssim_complex(x: torch.Tensor, y: torch.Tensor, kernel: torch.Tensor, data_r
     ssim_val = ssim_map.mean(1)
     cs = cs_map.mean(1)
 
-    if full:
-        return ssim_val, cs
-
-    return ssim_val
+    return ssim_val, cs
 
 
 def _multi_scale_ssim_complex(x: torch.Tensor, y: torch.Tensor, data_range: Union[int, float],
                               kernel: torch.Tensor, scale_weights_tensor: torch.Tensor, k1: float,
                               k2: float) -> torch.Tensor:
+    r"""Calculate Multi scale Structural Similarity (MS-SSIM) index for Complex X and Y.
+
+        Args:
+            x: Batch of complex images, (N,C,H,W,2).
+            y: Batch of complex images, (N,C,H,W,2).
+            data_range: Value range of input images (usually 1.0 or 255).
+            kernel: 2-D gauss kernel.
+            k1: Algorithm parameter, K1 (small constant, see [1]).
+            k2: Algorithm parameter, K2 (small constant, see [1]).
+                Try a larger K2 constant (e.g. 0.4) if you get a negative or NaN results.
+
+        Returns:
+            Value of Complex Mulri scale Structural Similarity (MS-SSIM) index.
+        """
     levels = scale_weights_tensor.size(0)
     min_size = (kernel.size(-1) - 1) * 2 ** (levels - 1) + 1
     if x.size(-2) < min_size or x.size(-3) < min_size:
         raise ValueError(f'Invalid size of the input images, expected at least {min_size}x{min_size}.')
-
     mcs = []
     ssim_val = None
-    for _ in range(levels):
-        ssim_val, cs = _ssim_per_channel_complex(x, y, kernel=kernel, data_range=data_range, k1=k1, k2=k2)
-
+    for iteration in range(levels):
         x_real = x[..., 0]
         x_imag = x[..., 1]
         y_real = y[..., 0]
         y_imag = y[..., 1]
-        mcs.append(cs)
+        if iteration > 0:
```

Comment:
The same question here: why do you need this condition?

Reply:
The same answer here :)